storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java

```diff
@@ -1183,11 +1188,12 @@ public void addNonCanonicalSidecar(final DataColumnSidecar sidecar) {
   }
 
   @Override
-  public void pruneAllSidecars(final UInt64 tillSlotInclusive) {
+  public void pruneAllSidecars(final UInt64 tillSlotInclusive, final int pruneLimit) {
     try (final Stream<DataColumnSlotAndIdentifier> prunableIdentifiers =
-            streamDataColumnIdentifiers(UInt64.ZERO, tillSlotInclusive);
+            streamDataColumnIdentifiers(UInt64.ZERO, tillSlotInclusive).limit(pruneLimit);
         final Stream<DataColumnSlotAndIdentifier> prunableNonCanonicalIdentifiers =
-            streamNonCanonicalDataColumnIdentifiers(UInt64.ZERO, tillSlotInclusive);
+            streamNonCanonicalDataColumnIdentifiers(UInt64.ZERO, tillSlotInclusive)
+                .limit(pruneLimit);
         final FinalizedUpdater updater = finalizedUpdater()) {
       prunableIdentifiers.forEach(updater::removeSidecar);
```

Comment:
It would be good to see here debug info with what was actually pruned, slot range

Reply:
added a debug log there